pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTType.java

```diff
@@ -49,7 +51,22 @@ public int getArrayDepth() {
         return 0; // this is not an array
     }
 
+
+    /**
+     *
+     * @deprecated Use {@link #isArrayType()}
+     */
+    @Deprecated
     public boolean isArray() {
         return getArrayDepth() > 0;
     }
+
+
+    /**
+     * Returns true if this type is an array type.
+     *
+     */
+    public boolean isArrayType() {
+        return isArray();
```

Comment:
The implementation of this new method now depends on a deprecated method... I usually use already the new implementation and delegate from the deprecated to the new one... (I know, we'll change the impl of `isArrayType()` in 7.0.0 to use the other AST nodes).

Reply:
You're right it should be the other way around